### PR TITLE
Use a virtualized list for logs

### PR DESCRIPTION
**Pull request by ccschmitz**
## Summary

Sets up [TanStack Virtual](https://tanstack.com/virtual/v3) to efficiently render a large list of logs. Also tweaks the loading state to make it easier to see when infinite scrolling.

## How did you test this change?

Local click test. Would also like to compare performance on the PR preview vs production to see if this feels like an improvement.

## Are there any deployment considerations?

N/A

### CODE REVIEW

File: frontend/src/pages/LogsPage/LogsTable/LogsTable.tsx [hunk 1]
```diff
@@ -137,17 +147,23 @@ export const LogsTable = ({ data, loading, query }: Props) => {
...
 						cssClass={clsx(styles.row, {
 							[styles.rowExpanded]: row.getIsExpanded(),
 						})}
-						key={row.id}
+						key={virtualRow.key}
+						data-index={virtualRow.index}
 						cursor="pointer"
 						onClick={row.getToggleExpandedHandler()}
-						mb="1"
+						mb="2"
```
Comment (ccschmitz) [inline frontend/src/pages/LogsPage/LogsTable/LogsTable.tsx:165]: A subtle design change requests by Julian.

**Comment by et** [comment]
Thanks for wiring this up! I'll take a look.

FYI - test failure is unrelated. @aptlin is looking into this.

**Comment by et** [comment]
Hey @ccschmitz - I'm not seeing forward scrolling on the render preview. I expect hitting the end of the window should load the nest results and I'm not seeing the next request fire.
Does this work locally for you?

**Comment by ccschmitz** [comment]
@et I'll look again when I'm back at the comp, but this was definitely working for me 🤔

**Comment by et** [comment]
I wonder if it has anything to do with my monitor size.
I'll set some time up to pair next week. Enjoy your weekend!

**Review by et — APPROVED** [review]
@ccschmitz and I paired on this. Looks like there might be an issue with the backend request timing out but it's not related to this PR.

**Pull request merged**